InnoGotchi.API/Infrastructure.Services/PetConditionService.cs

```diff
@@ -0,0 +1,218 @@
+                };
+                await _repositoryManager.FeedingHistoryRepository.CreateRecordAsync(feeding);
+
+                if (pet.HungerLevel == HungerLevel.Dead)
+                {
+                    pet.ThirstyLevel = ThirstyLevel.Dead;
+                    pet.DeathDay = feeding.ChangesDate;
+                    break;
+                }
+            }
+            while (thirstyDuration >= PetConfiguration.DrinkingFrequencyInHours)
+            {
+                thirstyDuration -= PetConfiguration.DrinkingFrequencyInHours;
+                pet.ThirstyLevel -= 1;
+
+                drinking = new ThirstyStateChanges()
+                {
+                    PetId = pet.Id,
+                    ChangesDate = drinking.ChangesDate.AddHours(PetConfiguration.DrinkingFrequencyInHours),
+                    ThirstyState = pet.ThirstyLevel
+                };
+                await _repositoryManager.DrinkingHistoryRepository.CreateRecordAsync(drinking);
+
+                if (pet.ThirstyLevel == ThirstyLevel.Dead)
+                {
+                    pet.HungerLevel = HungerLevel.Dead;
+                    if (pet.DeathDay > drinking.ChangesDate || pet.DeathDay == null)
+                        pet.DeathDay = drinking.ChangesDate;
+                    break;
+                }
+            }
+
+            await _repositoryManager.SaveAsync();
+
+            pet.HappynessDayCount = await CalculateHappynessDayCount(pet.Id);
+
+            await _repositoryManager.SaveAsync();
+
+            return pet;
+        }
+
+        public int CalculateAge(Pet pet) =>
+            pet.DeathDay != null ?
+                (pet.DeathDay - pet.Birthday).Value.Days / PetConfiguration.OnePetAgeInDays :
+                (DateTime.Now - pet.Birthday).Days / PetConfiguration.OnePetAgeInDays;
+
+        public async Task<double> CalculateHappynessDayCount(Guid petId)
+        {
+            var hungryRecords =
+                (await _repositoryManager.FeedingHistoryRepository.GetHistoryByPetIdAsync(petId, false)).ToList();
+            var thirstyRecords =
+                (await _repositoryManager.DrinkingHistoryRepository.GetHistoryByPetIdAsync(petId, false)).ToList();
+            hungryRecords.Add(new HungryStateChanges { PetId = petId, ChangesDate = DateTime.Now });
+            thirstyRecords.Add(new ThirstyStateChanges { PetId = petId, ChangesDate = DateTime.Now });
+
+            var startH = DateTime.Now; var endH = DateTime.Now;
+            var startT = DateTime.Now; var endT = DateTime.Now;
+
+            int i = 0, j = 0;
+
+            var count = 0.0;
+
+            bool emptyH = true; bool emptyT = true;
+            bool nextT = true, nextH = true;
+
+            while (true)
+            {
+                if (nextH)
+                {
+                    for (; i < hungryRecords.Count(); i++)
+                    {
+                        if (hungryRecords[i].HungerState >= HungerLevel.Normal)
+                        {
```

Comment:
too many levels of nesting, max is 2-3. You need to extract this to other methods. Please also notice that you can use foreach instead of for and select if necessary

Reply:
made